Taiko no Tatsujin Drum Controller (Tatacon) support

nullstalgia opened the pull request:
Re-opened since I accidentally deleted the old repo :(

Based on: [http://wiibrew.org/wiki/Wiimote/Extension_Controllers/TaTaCon](http://wiibrew.org/wiki/Wiimote/Extension_Controllers/TaTaCon)

[https://github.com/mon/TataconUSB](https://github.com/mon/TataconUSB)

Tested working! However, I did hear of a Wii-U drum that might have a different I2C address/format. But since I don't have $120 to spend on whats essentially a rumor, I'm choosing to ignore that.

Tony556 committed:
Add initial Taiko no Tatsujin Drum (Tatacon) support

Tony556 committed:
Tatacon Keywords

Tony556 committed:
Library Properties

Tony556 committed:
Spacing and printf fix

Tony556 committed:
Change debug title; Change byte used from 1st to 6th

dmadison commented:
Thanks for the controller request. You've got a few issues here. The function names don't match your keywords or the examples so this fails to compile. Your `specificInit` function also tries to set an invalid request size which you're then exceeding the range of with the data maps.

In the demo example, are the millis checks necessary? I'm not familiar with the game so I'm assuming the "large notes" are a game mechanic? Regardless the center/rim outputs are the same so should probably be combined with a logical OR. Those functions should also be stored to a boolean and then checked if `== true` for clarity.

Please also add the controller type to the keyword constants at the bottom for completeness, and add the controller to the README.

Tony556 committed:
Update Arduino library files, fix spacing and capitalization
Fix IdentifyController spacing
Add to keyword, readme, library.properties
Start to fix demo

nullstalgia commented:
Oh crap, I totally forgot about the Demo! That's 100% my bad, I apologize. I'll try to have more scrutiny in the future when submitting merges. (I was trying to get this done too quickly to make up for shipping delays...)

Most stated issues have been fixed, except for those relating to the demo's intended functionality. It does compile now, but I should wait for tomorrow to continue working on it. (Unless you think it's fine to go on without a demo for such a simple controller? Or do something much simpler?)

dmadison commented:
The demo is needed as most users won't dig through the source code to find the function names, though I think it could be simplified.

Clarify something for me though: your name is on this pull request and on these files for the `@author` and `@license` tags, but these commits are by the user @Tony556?
